Mimir.Worker/Scrapper/DiffScrapper.cs

```diff
@@ -34,6 +38,10 @@ public async Task ExecuteAsync(long baseIndex, long targetIndex)
                 currentBaseIndex,
                 currentTargetIndex
             );
+            var txs = await GetTransactions(
+                currentBaseIndex,
+                currentTargetIndex - currentBaseIndex
+            );
```

Comment:
You need to verify that these transactions are successful transactions (i.e., `TxExecution.Fail == false`). The `query.transaction.ncTransactions` query in NineChronicles.Headless GraphQL, doesn't return only successful transactions. So it needs to verify it. If it doesn't verify, I can assume a attack vector like the below steps:

1. There is *Attacker A*.
2. The *A* makes *Transaction T* that fails and has incorrect arguments. (e.g., Argument address should be `RuneSlotState`'s address but the address was `AvatarState`'s address → means *incorrect*)
3. The *T* has failed but Mimir doesn't verify it. So mimir may skip the changes because it fails to deserialize the state or may save polluted state if the state is compatible with the polluted states.
4. Some services uses Mimir, it cannot get latest state of the address or gets polluted state.

------

> the state is compatible with the polluted states

It means a situation that `A` state can be deserialized as `B` state too like the below code. (I write in TypeScript because I'm familiar with it recently. It is not valid TypeScript code, just pseudo-code)

```
class A {
  constructor(dict) {
    this.name = dict['n']
    this.level = dict['l']
    this.power = dict['p']
  }

  encode(): {
    dict = new Dict();
    dict['n'] = this.name;
    dict['l'] = this.level;
    dict['p'] = this.power;
    return dict;
  }
}

class B {
  constructor(dict) {
    this.no = dict['n']
    this.leverage = dict['l']
  }

  encode(): {
    dict = new Dict();
    dict['n'] = this.no;
    dict['l'] = this.leverage;
    return dict;
  }
}

const dict;
new B(new A(dict).encode()) // Successful.
```

Reply:
😱

Reply:
In that case... since the nodes need to perform validation, should we add this functionality to the headless?

Reply:
I don't think so. Since mimir fetches states as raw, headless cannot know it too. I think the below things are enough to prevent it. (maybe)

1. Check the transaction is successful.
2. Index address type hint correctly.

Reply:
Done! I added transaction result check step